src/utils/helpers/application/buildQueryFilters.ts

```diff
@@ -0,0 +1,119 @@
+  }, {})
+  return graphQLfilter
+}
+interface FilterMap {
+  [key: string]: (value: string) => object
+}
+
+const mapQueryToFilterField: FilterMap = {
+  type: (value: string) => {
+    return { templateCode: { equalToInsensitive: value } }
+  },
+  // category -- not yet implemented in schema
+  stage: (values: string) => {
+    return {
+      stage: inList(values),
+    }
+  },
+  status: (values: string) => {
+    return { status: inEnumList(values, ApplicationStatus) }
+  },
+  outcome: (values: string) => {
+    return { outcome: inEnumList(values, ApplicationOutcome) }
+  },
+  // action
+  // assigned
+  // consolidator
+  applicant: (values: string) => {
+    return {
+      or: [
+        { applicant: inList(values) },
+        { applicantFirstName: inList(values) },
+        { applicantLastName: inList(values) },
+        { applicantUsername: inList(values) },
+      ],
+    }
+  },
+  org: (values: string) => {
+    return { orgName: inList(values) }
+  },
+  lastActiveDate: (value: string) => {
+    const [startDate, endDate] = parseDateString(value)
+    console.log('Dates:', startDate, endDate)
+    return { lastActiveDate: { greaterThanOrEqualTo: startDate, lessThan: endDate } }
+  },
+  // deadlineDate (TBD)
+  search: (value: string) => {
+    return {
+      or: [
+        { name: { includesInsensitive: value } },
+        { applicantUsername: { includesInsensitive: value } },
+        { applicant: { includesInsensitive: value } },
+        { orgName: { includesInsensitive: value } },
+        { templateName: { includesInsensitive: value } },
+        { stage: { startsWithInsensitive: value } },
+      ],
+    }
+  },
+}
+
+const splitCommaList = (values: string) => values.split(',')
+
+// Use this if the values can be free text strings (e.g. stage name)
+const inList = (values: string) => {
+  return { inInsensitive: splitCommaList(values) }
+}
+
+// Use this if the values must conform to an Enum type (e.g. status, outcome)
+const inEnumList = (values: string, enumList: any) => {
+  return {
+    in: splitCommaList(values)
+      .map((value) => value.toUpperCase())
+      .filter((value) => {
+        return Object.values(enumList).includes(value)
+      }),
+  }
+}
+
+const parseDateString = (dateString: string) => {
+  if (mapNamedDates?.[dateString]) return mapNamedDates[dateString]
+  const [startDate, endDate] = dateString.split(':')
+  if (endDate === undefined)
+    // Exact date -- add 1 to cover until start of the next day
+    return [startDate, datePlusDays(1, startDate)]
+  if (endDate === '') return [startDate, undefined] // No end date boundary
+  if (startDate === '') return [undefined, endDate] // No start date boundary
+  return [startDate, endDate]
+}
+
+const datePlusDays = (offset = 0, dateString: string | undefined = undefined) => {
+  if (dateString) return DateTime.fromISO(dateString).plus({ days: offset }).toISODate()
+  return DateTime.local().plus({ days: offset }).toISODate()
+}
```

Comment:
I would move these utility methods to the top of the file. I think it's more clear.
```suggestion
const datePlusDays = (offset = 0, dateString: string | undefined = undefined) => {
  return dateString ? DateTime.fromISO(dateString).plus({ days: offset }).toISODate()
  : DateTime.local().plus({ days: offset }).toISODate()
}
```

Reply:
> I would move these utility methods to the top of the file. I think it's more clear.

Really? I thought we've generally been keeping these "helper" methods down the bottom so they don't get in the way of the main functionality. That's how we've been doing it in React files at least.